src/cfnlint/rules/resources/iam/Permissions.py

```diff
@@ -0,0 +1,145 @@
+        'AWS::SNS::TopicPolicy',
+        'AWS::SQS::QueuePolicy'
+    ]
+
+    def load_service_map(self):
+        """
+        Convert policies.json into a simpler version for more efficient key lookup.
+        """
+        service_map = cfnlint.helpers.load_resources('/data/AdditionalSpecs/Policies.json')['serviceMap']
+
+        policy_service_map = {}
+
+        for _, properties in service_map.items():
+            # Some services have the same name for different generations; like elasticloadbalancing.
+            if properties['StringPrefix'] in policy_service_map:
+                policy_service_map[properties['StringPrefix']] += properties['Actions']
+            else:
+                policy_service_map[properties['StringPrefix']] = properties['Actions']
+
+        return policy_service_map
+
+    def check_permissions(self, action, path, service_map):
+        """
+        Check if permission is valid
+        """
+        matches = []
+
+        if action == '*':
+            return matches
+
+        service, permission = action.split(':')
+
+        if service in service_map:
+            if permission.endswith('*'):
+                wilcarded_permission = permission.split('*')[0]
+                if not any(wilcarded_permission in action for action in service_map[service]):
+                    message = 'Invalid permission "{}" for "{}" found in permissions'
+                    matches.append(
+                        RuleMatch(
+                            path,
+                            message.format(permission, service, '/'.join(path))))
+
+            elif permission.startswith('*'):
+                wilcarded_permission = permission.split('*')[1]
+                if not any(wilcarded_permission in action for action in service_map[service]):
+                    message = 'Invalid permission "{}" for "{}" found in permissions'
+                    matches.append(
+                        RuleMatch(
+                            path,
+                            message.format(permission, service, '/'.join(path))))
+
+            elif permission not in service_map[service]:
+                message = 'Invalid permission "{}" for "{}" found in permissions'
+                matches.append(
+                    RuleMatch(
+                        path,
+                        message.format(permission, service, '/'.join(path))))
+        else:
+            message = 'Invalid service "{}" found in permissions'
+            matches.append(
+                RuleMatch(
+                    path,
+                    message.format(service, '/'.join(path))))
+
+        return matches
+
+    def get_actions(self, effective_permissions):
+        """return all actions from a statement"""
+
+        actions = []
+
+        if 'Action' in effective_permissions:
+            if isinstance(effective_permissions.get('Action'), str):
+                actions.append(effective_permissions.get('Action'))
+            else:
```

Comment:
This may be a weird one but planning for bad templates is hard to do.  Should we be making sure that this is actually a list before extending it as a list?